apps/sports-helsinki/src/domain/search/eventSearch/utils.tsx

```diff
@@ -427,6 +428,30 @@ export const getOrganizationSearchUrl = (
   );
 };
 
+export const getHelsinkiOnlySearchUrl = (
+  source: EventFields | Venue,
+  router: NextRouter,
+  locale: AppLanguage
+): string => {
+  if (source.__typename == 'Venue') {
+    return routerHelper.getLocalizedCmsItemUrl(
+      ROUTES.SEARCH,
+      { searchType: 'venue', helsinkiOnly: 'true' },
+      locale
+    );
+  } else if (source.__typename == 'EventDetails') {
+    return routerHelper.getLocalizedCmsItemUrl(
+      ROUTES.SEARCH,
+      {
+        searchType: source.typeId ?? EventTypeId.General,
+        helsinkiOnly: 'true',
+      },
+      locale
+    );
+  }
```

Comment:
Would it make sense to create new reusable functions to check the types?

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isEvent(obj: any): obj is EventFields {
  return (
    // eslint-disable-next-line no-underscore-dangle
    obj.__typename === 'EventDetails'
  );
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isVenue(obj: any): obj is Venue {
  return (
    // eslint-disable-next-line no-underscore-dangle
    obj.__typename === 'Venue'
  );
}

export const getHelsinkiOnlySearchUrl = (
  source: EventFields | Venue,
  router: NextRouter,
  locale: AppLanguage
): string => {
  if (isVenue(source)) {
    return routerHelper.getLocalizedCmsItemUrl(
      ROUTES.SEARCH,
      { searchType: 'venue', helsinkiOnly: 'true' },
      locale
    );
  } else if (isEvent(source)) {
    return routerHelper.getLocalizedCmsItemUrl(
      ROUTES.SEARCH,
      {
        searchType: source.typeId ?? EventTypeId.General,
        helsinkiOnly: 'true',
      },
      locale
    );
  }
  throw new Error('Invalid source type for getHelsinkiOnlySearchUrl');
};
```

Reply:
Taken into account in ea2b90da2f5c23a2b3a06d4586b31e6e964c9caa